include/aikido/robot/ConcreteManipulator.hpp

```diff
@@ -0,0 +1,148 @@
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path) override;
+
+  // Documentation inherited.
+  virtual void executeTrajectory(
+      const trajectory::TrajectoryPtr& trajectory) override;
+
+  // Documentation inherited.
+  /// \copydoc Robot::getNamedConfiguration
+  virtual boost::optional<Eigen::VectorXd> getNamedConfiguration(
+      const std::string& name) const override;
+
+  // Documentation inherited.
+  virtual void setNamedConfigurations(
+      std::unordered_map<std::string, const Eigen::VectorXd>
+          namedConfigurations) override;
+
+  // Documentation inherited.
+  virtual std::string getName() const override;
+
+  // Documentation inherited.
+  virtual dart::dynamics::MetaSkeletonPtr getMetaSkeleton() override;
+
+  // Documentation inherited.
+  virtual aikido::statespace::dart::MetaSkeletonStateSpacePtr getStateSpace()
+      override;
+
+  // Documentation inherited.
+  virtual void setRoot(Robot* robot) override;
+
+  // Documentation inherited.
+  virtual void step(
+      const std::chrono::system_clock::time_point& timepoint) override;
+
+  // Documentation inherited.
+  virtual aikido::constraint::CollisionFreePtr getSelfCollisionConstraint(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton) override;
+
+  // Documentation inherited.
+  virtual aikido::constraint::TestablePtr getFullCollisionConstraint(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const constraint::CollisionFreePtr& collisionFree) override;
+
+  /// Plans to a desired end-effector offset with fixed orientation.
+  /// \param[in] space The StateSpace for the metaskeleton.
+  /// \param[in] metaSkeleton Metaskeleton to plan with.
+  /// \param[in] body Bodynode for the end effector.
+  /// \param[in] collisionFree CollisionFree constraint to check.
+  /// Self-collision is checked by default.
+  /// \param[in] direction Direction unit vector in the world frame.
+  /// \param[in] distance Distance distance to move, in meters.
+  /// \param[in] timelimit Timelimit for planning
+  /// \return Output trajectory
+  trajectory::TrajectoryPtr planToEndEffectorOffset(
+      const statespace::dart::MetaSkeletonStateSpacePtr& space,
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const dart::dynamics::BodyNodePtr& body,
+      const constraint::CollisionFreePtr& collisionFree,
+      const Eigen::Vector3d& direction,
+      double distance,
+      double timelimit,
+      double positionTolerance,
+      double angularTolerance);
+
+  /// Returns the direction of an end effector (along z axis) in the world frame
+  /// \param body Bodynode for the end effector
+  /// \return The direction of the end effector (z axis of the frame)
+  Eigen::Vector3d getEndEffectorDirection(
```

Comment:
If I remember correctly, this is mostly a helper function so I'm not sure if it should be exposed as part of the public API. Can we make it protected, or a function in `aikido::common`?